config/main.py

```diff
@@ -1333,6 +1344,369 @@ def del_vlan_dhcp_relay_destination(ctx, vid, dhcp_relay_destination_ip):
     else:
         ctx.fail("{} is not a DHCP relay destination for {}".format(dhcp_relay_destination_ip, vlan_name))
 
+# Validate VLAN range.
+#
+def vlan_range_validate(ctx, vid1, vid2):
+    vlan1 = 'Vlan{}'.format(vid1)
+    vlan2 = 'Vlan{}'.format(vid2)
+
+    if vlan_id_is_valid(vid1) is False:
+        ctx.fail("{} is not within allowed range of 1 through 4094".format(vlan1))
+    if vlan_id_is_valid(vid2) is False:
+        ctx.fail("{} is not within allowed range of 1 through 4094".format(vlan2))
+
+    if vid2 <= vid1:
+        ctx.fail(" vid2 should be greater than vid1")
+
+#
+# Return a string with ranges separated by hyphen.
+#
+def get_hyphenated_string(vlan_list):
+    vlan_list.sort()
+    G = (list(x) for _,x in groupby(vlan_list, lambda x,c=count(): next(c)-x))
+    hyphenated_string = ",".join("-".join(map(str,(g[0],g[-1])[:len(g)])) for g in G)
+    return hyphenated_string
+
+#
+# 'range' group ('config vlan range ...')
+#
+@vlan.group('range')
+@click.pass_context
+def vlan_range(ctx):
+    """VLAN-range related configuration tasks"""
+    pass
+
+@vlan_range.command('add')
+@click.argument('vid1', metavar='<vid1>', required=True, type=int)
+@click.argument('vid2', metavar='<vid2>', required=True, type=int)
+@click.option('-w', "--warning", is_flag=True, help='warnings are not suppressed')
+@click.pass_context
+def add_vlan_range(ctx, vid1, vid2, warning):
+    db = ctx.obj['db']
+
+    vlan_range_validate(ctx, vid1, vid2)
+
+    vid2 = vid2+1
+
+    warning_vlans_list = []
+    curr_vlan_count = 0
+    config_db = ConfigDBConnector()
+    config_db.connect()
+    clients = config_db.get_redis_client(config_db.CONFIG_DB)
+    pipe = clients.pipeline()
+    for vid in range (vid1, vid2):
+        vlan = 'Vlan{}'.format(vid)
+
+        if len(db.get_entry('VLAN', vlan)) != 0:
+            if warning is True:
+                warning_vlans_list.append(vid)
+            continue
+
+        pipe.hmset('VLAN|{}'.format(vlan),  {'vlanid': vid})
```

Comment:
need use swsssdk as abstract layer of API. using redis client api make hard to support multi-db implementation in the future.

Reply:
@lguohan if I understood correctly, should I add a new API in swss-common/swsssdk and call the API from here instead of using redis client API directly?

Reply:
Anil - he's asking you to look in https://github.com/Azure/sonic-swss-common for a suitable DB access function and use this. If you don't find one that meets your needs then please add it - thanks.